Essay_Analysis_Tool/Business/MainViewController.cs

```diff
@@ -0,0 +1,227 @@
+            _view.UpdateDocumentMap();
+            _view.HighlightCurrentLine();
+        }
+
+        public bool Save(FATabStripItem tab)
+        {
+            if (tab == null)
+            {
+                _view.Log(Resources.NullTabStripItem, LoggerMessageType.Error);
+                MessageBox.Show("Save Unsucessfull.", "Save not complete", MessageBoxButtons.OK, MessageBoxIcon.Error);
+                return false;
+            }
+
+            FastColoredTextBox tb = tab.Controls[0] as FastColoredTextBox;
+
+            if (tab.Tag == null)
+            {
+                tab = _view.CallSaveFileDialog(tab);
+                if (tab == null)
+                {
+                    return true;
+                }
+            }
+
+            string filePath = (string)tab.Tag;
+
+            SetCurrentEditorSyntaxHighlight(filePath, tb);
+
+            File.WriteAllText(filePath, tb.Text);
+            _view.UpdateChangedFlag(false);
+
+            return true;
+        }
+
+        public void SetCurrentEditorSyntaxHighlight(string fName, FastColoredTextBox mainEditor)
+        {
+            char[] name = fName.ToCharArray();
+            string ext = "";
+            int token = fName.Length - 1;
+
+            while (name[token] != '.')
+            {
+                ext += name[token].ToString();
+                token -= 1;
+            }
+            name = ext.ToCharArray();
+            Array.Reverse(name);
+            token = 0;
+            ext = "";
+            while (token < name.Length)
+            {
+                ext += name[token].ToString();
+                token += 1;
+            }
+
+            switch (ext)
+            {
+                case _html:
+                    ChangeSyntax(mainEditor, Language.HTML);
+                    break;
+                case _xml:
+                    ChangeSyntax(mainEditor, Language.XML);
+                    break;
+                case _javascript:
+                    ChangeSyntax(mainEditor, Language.JS);
+                    break;
+                case _lua:
+                    ChangeSyntax(mainEditor, Language.Lua);
+                    break;
+                case _csharp:
+                    ChangeSyntax(mainEditor, Language.CSharp);
+                    break;
+                case _sql:
+                    ChangeSyntax(mainEditor, Language.SQL);
+                    break;
+                case _vb:
+                    ChangeSyntax(mainEditor, Language.VB);
+                    break;
+                case _vbs:
+                    ChangeSyntax(mainEditor, Language.VB);
+                    break;
+                default:
+                    mainEditor.Language = Language.Custom;
+                    break;
+            }
+        }
+
+        public void ChangeSyntax(FastColoredTextBox tb, Language language)
+        {
+            if (tb == null)
+            {
+                _view.Log(Resources.InvalidArgument, LoggerMessageType.Error);
```

Comment:
This should be something like `_logViewControll.AddError(Resources.InvalidArgument);` and the view would be invoked from the `_logViewControll.AddError()` method.